aws/request/request.go

```diff
@@ -223,26 +222,36 @@ func (r *Request) Sign() error {
 // be executed in the order they were set.
 func (r *Request) Send() error {
 	for {
-		r.Sign()
-		if r.Error != nil {
-			return r.Error
-		}
-
 		if aws.BoolValue(r.Retryable) {
 			if r.Config.LogLevel.Matches(aws.LogDebugWithRequestRetries) {
 				r.Config.Logger.Log(fmt.Sprintf("DEBUG: Retrying Request %s/%s, attempt %d",
 					r.ClientInfo.ServiceName, r.Operation.Name, r.RetryCount))
 			}
-
+			body := r.HTTPRequest.Body.(*offsetReader)
```

Comment:
Need to do runtime type assertion here and only call `CloseCopy` if the Body is a `offsetReader`